### PR TITLE
BL602 pull request

### PR DESCRIPTION
**Pull request by jczhang777**
#### Problem
* Add new platform for matter

#### Change overview
Add chip bl602 for matter

#### Testing
How was this tested? (at least one bullet point required)
* see "connectedhomeip/examples/lighting-app/bouffalolab/bl602/README.md"

### CODE REVIEW

**Comment by andy31415** [comment]
@jczhang777 is this the same as #16358 ?

**Review by andy31415 — DISMISSED** [review]
There seems to be some reviews lost in history as the same pull request is being created over and over again: https://github.com/project-chip/connectedhomeip/pulls?q=is%3Apr+bl602

See comments in #16283 - need build_examples support.

Also for history and review please keep the same PR and do not open new PRs for every change as that is very spammy

**Comment by jczhang777** [comment]
> There seems to be some reviews lost in history as the same pull request is being created over and over again: https://github.com/project-chip/connectedhomeip/pulls?q=is%3Apr+bl602
> 
> See comments in #16283 - need build_examples support.
> 
> Also for history and review please keep the same PR and do not open new PRs for every change as that is very spammy

sorry about that.

**Comment by andy31415** [comment]
> > There seems to be some reviews lost in history as the same pull request is being created over and over again: https://github.com/project-chip/connectedhomeip/pulls?q=is%3Apr+bl602
> > See comments in #16283 - need build_examples support.
> > Also for history and review please keep the same PR and do not open new PRs for every change as that is very spammy
> 
> sorry about that. i had fixed issues that #16283 mentioned。but i do not know why spell check failed in README.md. build_examples support is changed as you comments.

for build_examples.py to work I expect changes in `scripts/build/build/targets.py` as well as adding a new platform  and unit test updates. I do not see these in this PR.

**Review by andy31415 — COMMENTED** [review]
Please remove README.pdf

git does not like binary files, we have a readme.md

File: examples/lighting-app/bouffalolab/bl602/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+-   Install some additional tools(likely already present for CHIP developers):
+
+    #Linux \$ sudo apt-get install git libwebkitgtk-1.0-0 ninja-build
+
+    #Mac OS X \$ brew install ninja
+
+*   Build the example application:
+
+```
+connectedhomeip$ ./scripts/examples/gn_bl602_example.sh ./examples/lighting-app/bouffalolab/bl602 out/lighting_app_bl602
```
Comment (andy31415) [inline examples/lighting-app/bouffalolab/bl602/README.md:14]: I expect this to be like:

```
./scripts/build/build_examples.py --target bouffalolab-bl602-lighting-app build
```

**Comment by jczhang777** [comment]
> Restyler complains just about one empty line:
> 
> [380f3e1](https://github.com/project-chip/connectedhomeip/commit/380f3e1129de71d06af61816738292194c01fde1)

very thankful.

**Comment by jczhang777** [comment]
> Please remove README.pdf
> 
> git does not like binary files, we have a readme.md

ok.

**Comment by jczhang777** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

this project is on going.

**Review by woody-apple — APPROVED** [review]
Fast tracking platform changes.

**Comment by jczhang777** [comment]
> Fast tracking platform changes.



> /rebase

working on it. thanks.

> Fast tracking platform changes.

ok, I am working on it.

**Comment by jczhang777** [comment]
@andy31415 Hi, I had rebased my code, and fix some problems.Include https://github.com/project-chip/connectedhomeip/pull/16370#discussion_r829411785 that you had mentioned。

**Comment by jczhang777** [comment]
Hi, I had rebased my code, and fix some problems.Include https://github.com/project-chip/connectedhomeip/pull/16370#discussion_r829411785 that you had mentioned。Are there any other issues that I need to fix? very thankful。

**Comment by jczhang777** [comment]
@andy31415 Hi, I had rebased my code, and fix some problems.Include https://github.com/project-chip/connectedhomeip/pull/16370#discussion_r829411785 that you had mentioned。Are there any other issues that I need to fix? very thankful。

**Comment by bzbarsky-apple** [comment]
> Are there any other issues that I need to fix

The restyle failure.  See the automatically created commits in https://github.com/project-chip/connectedhomeip/pull/16921 for the needed changes.

**Comment by jczhang777** [comment]
> > Are there any other issues that I need to fix
> 
> The restyle failure. See the automatically created commits in #16921 for the needed changes.



> > Are there any other issues that I need to fix
> 
> The restyle failure. See the automatically created commits in #16921 for the needed changes.
I just fixed it. If there is anything else I need to modify, please let me know. thank you very much. 


> > Are there any other issues that I need to fix
> 
> The restyle failure. See the automatically created commits in #16921 for the needed changes.
I just fixed it. If there is anything else I need to modify, please let me know. thank you very much.

**Comment by jczhang777** [comment]
> > Are there any other issues that I need to fix
> 
> The restyle failure. See the automatically created commits in #16921 for the needed changes.

I just fixed it. If there is anything else I need to modify, please let me know. thank you very much.

**Comment by bzbarsky-apple** [comment]
For some reason CI did not kick off properly there.. You might need to force-push again.  :(

**Comment by jczhang777** [comment]
> For some reason CI did not kick off properly there.. You might need to force-push again. :(

I have pushed forcibly pushed up. Please check if there is anything else to modify. Thank you.

**Pull request merged**